File: exercism/go/binary-search/binary_search.go
package binarysearch

func SearchInts(list []int, key int) int {
	if len(list) == 0 {
		return -1
	}
	l, r := 0, len(list)-1
	for l != r {
		m := (l + r) / 2
		if key > list[m] {
			l = m + 1
		} else {
			r = m
		}
	}
	if list[l] == key {
		return l
	}
	return -1
}

File: exercism/zig/binary-search/binary_search.zig
pub fn binarySearch(comptime T: type, target: T, items: []const T) ?usize {
    if (items.len == 0) return null;
    var l: usize = 0;
    var r: usize = items.len - 1;
    while (l != r) {
        const m = (l + r + 1) / 2;
        if (target >= items[m]) l = m else r = m - 1;
    }
    if (items[l] == target) return l;
    return null;
}
